streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -1499,6 +1501,83 @@ protected StreamTask createStreamTask(final TaskId id, final Collection<TopicPar
         assertFalse(testStreamTask.committed);
     }
 
+    @Test
+    public void testThreadMetadataUpdate() {
```

Comment:
Can we update the test name? We prefer a descriptive name like `shouldDoSomethingOnMethod`

The test seems to be quite complex, too. Can we break it down in multiple smaller tests?